drivers/include/bmp180.h

```diff
@@ -0,0 +1,129 @@
+ */
+
+#ifndef BMP180_H_
+#define BMP180_H_
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name Oversampling modes
+ * @{
+ */
+#define BMP180_ULTRALOWPOWER          (0)
+#define BMP180_STANDARD               (1)
+#define BMP180_HIGHRES                (2)
+#define BMP180_ULTRAHIGHRES           (3)
+/** @} */
+
+/**
+ * @brief Calibration struct for the BMP180 sensor
+ */
+typedef struct {
+    int16_t ac1;          /**< ac1 coefficient */
+    int16_t ac2;          /**< ac2 coefficient */
+    int16_t ac3;          /**< ac3 coefficient */
+    int16_t b1;           /**< b1 coefficient */
+    int16_t b2;           /**< b2 coefficient */
+    int16_t mb;           /**< mb coefficient */
+    int16_t mc;           /**< mc coefficient */
+    int16_t md;           /**< md coefficient */
+    uint16_t ac4;         /**< ac4 coefficient */
+    uint16_t ac5;         /**< ac5 coefficient */
+    uint16_t ac6;         /**< ac6 coefficient */
+} bmp180_calibration_t;
+
+/**
+ * @brief Device descriptor for the BMP180 sensor
+ */
+typedef struct {
+    i2c_t i2c_dev;                     /**< I2C device which is used */
+    bmp180_calibration_t calibration;  /**< Device calibration */
+    uint8_t oversampling;              /**< Oversampling mode */
+} bmp180_t;
+
+/**
+ * @brief Initialize the given BMP180 device
+ *
+ * @param[out] dev          Initialized device descriptor of BMP180 device
+ * @param[in]  i2c          I2C bus the sensor is connected to
+ * @param[in]  mode         BMP180 oversampling mode
+ *
+ * @return                  0 on success
+ * @return                  -1 if given I2C is not enabled in board config
+ */
+int bmp180_init(bmp180_t *dev, i2c_t i2c, uint8_t mode);
+
+/**
+ * @brief Read temperature value from the given BMP180 device, returned in d°C
+ *
+ * @param[in] dev           Device descriptor of BMP180 device to read from
+ * @param[out] temperature  Temperature in d°C
+ *
+ * @return                  0 on success
+ * @return                  -1 if device's I2C is not enabled in board config
+ */
+int bmp180_read_temperature(bmp180_t *dev, int32_t *temperature);
+
+/**
+ * @brief Read pressure value from the given BMP180 device, returned in Pa
+ *
+ * @param[in]  dev          Device descriptor of BMP180 device to read from
+ * @param[out] pressure     Pressure in Pa
+ *
+ * @return                  0 on success
+ * @return                  -1 if device's I2C is not enabled in board config
+ */
+int bmp180_read_pressure(bmp180_t *dev, int32_t *pressure);
+
+/**
+ * @brief Compute altitude, returned in m.
+ *
+ * @param[in]  dev          Device descriptor of BMP180 device to read from
+ * @param[in]  pressure_0   The pressure at sea level in Pa
+ * @param[out] altitude     Altitude in m
+ *
+ * @return                  0 on success
+ * @return                  -1 if device's I2C is not enabled in board config
+ */
+int bmp180_altitude(bmp180_t *dev, double *pressure_0, double *altitude);
```

Comment:
Ok, altitude precision can be 1m, under I think it doesn't really make sense. I changed the returned type from `double` to `int32_t`.